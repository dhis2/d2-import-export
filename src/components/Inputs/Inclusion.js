import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { RadioGroupField } from '../index'

const inclusionOptions = [
    {
        value: 'SELECTED',
        label: i18n.t('Only include selected organisation unit'),
    },
    {
        value: 'CHILDREN',
        label: i18n.t('Include the first level of units inside selections'),
    },
    {
        value: 'DESCENDANTS',
        label: i18n.t('Include all units inside selections'),
    },
]
const defaultInclusionOption = inclusionOptions[0].value

const NAME = 'inclusion'
const DATATEST = 'input-inclusion'
const LABEL = i18n.t(
    'How should organisation units inside the selections be included?'
)

const Inclusion = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={inclusionOptions}
        dataTest={DATATEST}
        vertical
    />
)

export { Inclusion, defaultInclusionOption }
