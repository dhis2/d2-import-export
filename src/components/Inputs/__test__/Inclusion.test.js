import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Inclusion } from '../Inclusion'

jest.mock('react-final-form')

describe('Input component - Inclusion', () => {
    it('should render correctly', () => {
        const file = shallow(<Inclusion />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
