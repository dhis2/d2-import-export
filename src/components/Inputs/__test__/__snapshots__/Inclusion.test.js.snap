// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Inclusion should render correctly 1`] = `
<Field>
  <Label>
    Inclusion
  </Label>
  <RadioGroup
    label="Inclusion"
    name="inclusion"
    options={
      Array [
        Object {
          "label": "Selected organisation unit",
          "value": "selected",
        },
        Object {
          "label": "Include children of organisation unit",
          "value": "children",
        },
        Object {
          "label": "Include descendants of organisation unit",
          "value": "descendants",
        },
      ]
    }
  />
</Field>
`;
